components/console/single_command_tool.rst

```diff
@@ -6,68 +6,52 @@ Building a single Command Application
 
 When building a command line tool, you may not need to provide several commands.
 In such case, having to pass the command name each time is tedious. Fortunately,
-it is possible to remove this need by extending the application::
-
-    namespace Acme\Tool;
-
-    use Symfony\Component\Console\Application;
-    use Symfony\Component\Console\Input\InputInterface;
-
-    class MyApplication extends Application
-    {
-        /**
-         * Gets the name of the command based on input.
-         *
-         * @param InputInterface $input The input interface
-         *
-         * @return string The command name
-         */
-        protected function getCommandName(InputInterface $input)
-        {
-            // This should return the name of your command.
-            return 'my_command';
-        }
-
-        /**
-         * Gets the default commands that should always be available.
-         *
-         * @return array An array of default Command instances
-         */
-        protected function getDefaultCommands()
-        {
-            // Keep the core default commands to have the HelpCommand
-            // which is used when using the --help option
-            $defaultCommands = parent::getDefaultCommands();
+it is possible to remove this need declaring a single command application::
+
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
+
+  use Symfony\Component\Console\Application;
+  use Symfony\Component\Console\Input\InputArgument;
+  use Symfony\Component\Console\Input\InputInterface;
+  use Symfony\Component\Console\Input\InputOption;
+  use Symfony\Component\Console\Output\OutputInterface;
+
+  (new Application('echo', '1.0.0'))
+    ->register('echo')
+        ->addArgument('foo', InputArgument::OPTIONAL, 'The directory', 'foo')
+        ->addOption('bar', null, InputOption::VALUE_REQUIRED, 'Foobar', 'bar')
+        ->setCode(function(InputInterface $input, OutputInterface $output) {
+            $output->writeln('start');
+            $output->writeln($input->getArgument('foo'));
+            $output->writeln($input->getOption('bar'));
+        })
+    ->getApplication()
+    ->setDefaultCommand('echo', true)
```

Comment:
You're right. How about adding a comment like:

``setDefaultCommand`` function now accepts a second argument. If true, the application
is considerated as a Single command application.

ping @wouterj as my english is just so bad :)